Compilation error when USE_XSIMD defined

```python
#pythran export sigmoid(float32[:,:,:])
def sigmoid(z):
    return 1 / (1 + np.exp(-z))


#pythran export conv(float32[:,:], float32[:,:], int, int, int)
def conv(conv_matrix, qcnn_filter, length, batch_size, state_size):
    state_size_2 = state_size * 2
    state_size_3 = state_size * 3
    state_size_4 = state_size * 4

    conv_results = np.dot(conv_matrix, qcnn_filter).reshape(length, batch_size, state_size_4)
    conv_results[:, :, :state_size] = np.tanh(conv_results[:, :, :state_size])
    conv_results[:, :, state_size:state_size_4] = sigmoid(conv_results[:, :, state_size:state_size_4])

    state_results = np.zeros((batch_size, length, state_size_2), dtype=np.float32)
    state = np.zeros((batch_size, state_size), dtype=np.float32)
    for i in range(length):
        z = conv_results[i, :, :state_size]
        f = conv_results[i, :, state_size:state_size_2]
        o = conv_results[i, :, state_size_2:state_size_3]
        state = f * state + (1 - f) * z
        state_results[:, i, :state_size] = state * o

    state = np.zeros((batch_size, state_size), dtype=np.float32)
    for i in range(length - 1, -1, -1):
        z = conv_results[i, :, :state_size]
        f = conv_results[i, :, state_size_3:state_size_4]
        o = conv_results[i, :, state_size_2:state_size_3]
        state = f * state + (1 - f) * z
        state_results[:, i, state_size:] = state * o
    state_results = state_results.reshape((batch_size * length, state_size_2))
    return state_results
```

Without `-DUSE_XSIMD` it works fine, with `-DUSE_XSIMD` the generated cpp source code does not compile:

```
In file included from /Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/core.hpp:29:
In file included from /Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/types/slice.hpp:5:
In file included from /Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/types/NoneType.hpp:8:
In file included from /Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/__builtin__/bool_.hpp:7:
In file included from /Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/types/tuple.hpp:12:
In file included from /Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/types/ndarray.hpp:16:
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/utils/broadcast_copy.hpp:135:20: error: no viable conversion from 'decltype(xsimd::load_unaligned(this->data))' (aka 'xsimd::batch<double, 2>') to 'const xsimd::simd_type<typename numpy_gexpr<numpy_iexpr<const numpy_iexpr<const
      ndarray<float, pshape<long, long, long> > &> &>, contiguous_normalized_slice>::dtype>' (aka 'const batch<float, 4>')
        iter.store(*oiter);
                   ^~~~~~
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/utils/broadcast_copy.hpp:169:14: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::utils::vbroadcast_copy<(anonymous namespace)::pythonic::types::vectorizer_nobroadcast,
      (anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &> &>,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice> &, (anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long> > >' requested here
      return vbroadcast_copy<types::vectorizer_nobroadcast>(
             ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/utils/broadcast_copy.hpp:194:9: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::utils::_broadcast_copy<(anonymous namespace)::pythonic::types::vectorizer_nobroadcast, 1,
      0>::operator()<(anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &> &>,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice> &, (anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long> > >' requested here
        _broadcast_copy<types::vectorizer_nobroadcast, N, D>{}(self, other);
        ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/utils/broadcast_copy.hpp:204:7: note: in instantiation of member function '(anonymous namespace)::pythonic::utils::broadcast_copy_dispatcher<(anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous
      namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &> &>, (anonymous namespace)::pythonic::types::contiguous_normalized_slice> &,
      (anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long> >, 1, 0, true>::operator()' requested here
      broadcast_copy_dispatcher<E, F, N, D, vector_form>{}(self, other);
      ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/types/numpy_gexpr.hpp:383:21: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::utils::broadcast_copy<(anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous
      namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &> &>, (anonymous namespace)::pythonic::types::contiguous_normalized_slice> &,
      (anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long> >, 1, 0, true>' requested here
      return utils::broadcast_copy<
                    ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/types/numpy_gexpr.hpp:416:12: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::numpy_iexpr<const
      (anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &> &>, (anonymous namespace)::pythonic::types::contiguous_normalized_slice>::_copy<(anonymous
      namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::div, (anonymous namespace)::pythonic::types::broadcast<float, float>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::add,
      (anonymous namespace)::pythonic::types::broadcast<float, float>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::numpy::functor::exp, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::neg,
      (anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &> &>,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice> > > > > >' requested here
    return _copy(expr);
           ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:1701:19: note: (skipping 11 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
        *__result = *__first;
                  ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/utils/broadcast_copy.hpp:204:7: note: in instantiation of member function '(anonymous namespace)::pythonic::utils::broadcast_copy_dispatcher<(anonymous namespace)::pythonic::types::numpy_gexpr<const
      (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &, (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice> &, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::div, (anonymous namespace)::pythonic::types::broadcast<float, long>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous
      namespace)::pythonic::operator_::functor::add, (anonymous namespace)::pythonic::types::broadcast<float, long>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::numpy::functor::exp, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous
      namespace)::pythonic::operator_::functor::neg, (anonymous namespace)::pythonic::types::numpy_gexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &, (anonymous namespace)::pythonic::types::contiguous_normalized_slice,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice> > > > >, 3, 0, false>::operator()' requested here
      broadcast_copy_dispatcher<E, F, N, D, vector_form>{}(self, other);
      ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/types/numpy_gexpr.hpp:389:21: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::utils::broadcast_copy<(anonymous namespace)::pythonic::types::numpy_gexpr<const
      (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &, (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice> &, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::div, (anonymous namespace)::pythonic::types::broadcast<float, long>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous
      namespace)::pythonic::operator_::functor::add, (anonymous namespace)::pythonic::types::broadcast<float, long>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::numpy::functor::exp, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous
      namespace)::pythonic::operator_::functor::neg, (anonymous namespace)::pythonic::types::numpy_gexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &, (anonymous namespace)::pythonic::types::contiguous_normalized_slice,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice> > > > >, 3, 0, false>' requested here
      return utils::broadcast_copy < numpy_gexpr &, E, value,
                    ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/types/numpy_gexpr.hpp:416:12: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::types::numpy_gexpr<const (anonymous namespace)::pythonic::types::ndarray<float,
      (anonymous namespace)::pythonic::types::pshape<long, long, long> > &, (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice>::_copy<(anonymous
      namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::div, (anonymous namespace)::pythonic::types::broadcast<float, long>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::add,
      (anonymous namespace)::pythonic::types::broadcast<float, long>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::numpy::functor::exp, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::neg,
      (anonymous namespace)::pythonic::types::numpy_gexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &, (anonymous namespace)::pythonic::types::contiguous_normalized_slice,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice> > > > > >' requested here
    return _copy(expr);
           ^
NERRNN/speedup.cpp:226:260: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::types::numpy_gexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice>::operator=<(anonymous namespace)::pythonic::types::numpy_expr<(anonymous
      namespace)::pythonic::operator_::functor::div, (anonymous namespace)::pythonic::types::broadcast<float, long>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::add, (anonymous namespace)::pythonic::types::broadcast<float, long>,
      (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::numpy::functor::exp, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::neg, (anonymous namespace)::pythonic::types::numpy_gexpr<const
      (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &, (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice> > > > > >' requested here
    conv_results(pythonic::types::contiguous_slice(pythonic::__builtin__::None,pythonic::__builtin__::None),pythonic::types::contiguous_slice(pythonic::__builtin__::None,pythonic::__builtin__::None),pythonic::types::contiguous_slice(state_size,state_size_4)) = (pythonic::operator_::div(1L, (pythonic::operator...
                                                                                                                                                                                                                                                                   ^
NERRNN/speedup.cpp:266:69: note: in instantiation of function template specialization '__pythran_speedup::conv::operator()<(anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long> > &, (anonymous namespace)::pythonic::types::ndarray<float,
      (anonymous namespace)::pythonic::types::pshape<long, long> > &, long &, long &, long &>' requested here
                                auto res = __pythran_speedup::conv()(conv_matrix, qcnn_filter, length, batch_size, state_size);
                                                                    ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/xsimd/types/xsimd_sse_float.hpp:79:11: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'decltype(xsimd::load_unaligned(this->data))' (aka 'xsimd::batch<double, 2>') to
      'const xsimd::batch<float, 4> &' for 1st argument
    class batch<float, 4> : public simd_batch<batch<float, 4>>
          ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/xsimd/types/xsimd_sse_float.hpp:79:11: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'decltype(xsimd::load_unaligned(this->data))' (aka 'xsimd::batch<double, 2>') to
      'xsimd::batch<float, 4> &&' for 1st argument
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/xsimd/types/xsimd_sse_float.hpp:248:29: note: candidate constructor not viable: no known conversion from 'decltype(xsimd::load_unaligned(this->data))' (aka 'xsimd::batch<double, 2>') to 'const __m128 &' for 1st argument
    inline batch<float, 4>::batch(const __m128& rhs)
                            ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/xsimd/types/xsimd_sse_double.hpp:95:9: note: candidate function
        operator __m128d() const;
        ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/include/types/nditerator.hpp:186:59: note: passing argument to parameter 'val' here
    void store(xsimd::simd_type<typename E::dtype> const &);
                                                          ^
In file included from NERRNN/speedup.cpp:1:
In file included from /Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/core.hpp:29:
In file included from /Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/types/slice.hpp:5:
In file included from /Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/types/NoneType.hpp:8:
In file included from /Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/__builtin__/bool_.hpp:7:
In file included from /Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/types/tuple.hpp:12:
In file included from /Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/types/ndarray.hpp:16:
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/utils/broadcast_copy.hpp:135:20: error: no viable conversion from 'decltype(xsimd::load_unaligned(this->data))' (aka 'xsimd::batch<double, 2>') to 'const xsimd::simd_type<typename numpy_gexpr<numpy_iexpr<const numpy_iexpr<const
      ndarray<float, pshape<long, long, long> > &> &>, contiguous_normalized_slice>::dtype>' (aka 'const batch<float, 4>')
        iter.store(*oiter);
                   ^~~~~~
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/utils/broadcast_copy.hpp:161:14: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::utils::vbroadcast_copy<(anonymous namespace)::pythonic::types::vectorizer,
      (anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &> &>,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice> &, (anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long> > >' requested here
      return vbroadcast_copy<types::vectorizer>(std::forward<E>(self), other);
             ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/utils/broadcast_copy.hpp:196:9: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::utils::_broadcast_copy<(anonymous namespace)::pythonic::types::vectorizer, 1, 0>::operator()<(anonymous
      namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &> &>,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice> &, (anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long> > >' requested here
        _broadcast_copy<types::vectorizer, N, D>{}(self, other);
        ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/utils/broadcast_copy.hpp:204:7: note: in instantiation of member function '(anonymous namespace)::pythonic::utils::broadcast_copy_dispatcher<(anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous
      namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &> &>, (anonymous namespace)::pythonic::types::contiguous_normalized_slice> &,
      (anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long> >, 1, 0, true>::operator()' requested here
      broadcast_copy_dispatcher<E, F, N, D, vector_form>{}(self, other);
      ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/types/numpy_gexpr.hpp:383:21: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::utils::broadcast_copy<(anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous
      namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &> &>, (anonymous namespace)::pythonic::types::contiguous_normalized_slice> &,
      (anonymous namespace)::pythonic::types::ndarray<double, (anonymous namespace)::pythonic::types::pshape<long> >, 1, 0, true>' requested here
      return utils::broadcast_copy<
                    ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/types/numpy_gexpr.hpp:416:12: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::numpy_iexpr<const
      (anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &> &>, (anonymous namespace)::pythonic::types::contiguous_normalized_slice>::_copy<(anonymous
      namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::div, (anonymous namespace)::pythonic::types::broadcast<float, float>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::add,
      (anonymous namespace)::pythonic::types::broadcast<float, float>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::numpy::functor::exp, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::neg,
      (anonymous namespace)::pythonic::types::numpy_gexpr<(anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::numpy_iexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &> &>,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice> > > > > >' requested here
    return _copy(expr);
           ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/algorithm:1701:19: note: (skipping 11 contexts in backtrace; use -ftemplate-backtrace-limit=0 to see all)
        *__result = *__first;
                  ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/utils/broadcast_copy.hpp:204:7: note: in instantiation of member function '(anonymous namespace)::pythonic::utils::broadcast_copy_dispatcher<(anonymous namespace)::pythonic::types::numpy_gexpr<const
      (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &, (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice> &, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::div, (anonymous namespace)::pythonic::types::broadcast<float, long>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous
      namespace)::pythonic::operator_::functor::add, (anonymous namespace)::pythonic::types::broadcast<float, long>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::numpy::functor::exp, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous
      namespace)::pythonic::operator_::functor::neg, (anonymous namespace)::pythonic::types::numpy_gexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &, (anonymous namespace)::pythonic::types::contiguous_normalized_slice,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice> > > > >, 3, 0, false>::operator()' requested here
      broadcast_copy_dispatcher<E, F, N, D, vector_form>{}(self, other);
      ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/types/numpy_gexpr.hpp:389:21: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::utils::broadcast_copy<(anonymous namespace)::pythonic::types::numpy_gexpr<const
      (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &, (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice> &, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::div, (anonymous namespace)::pythonic::types::broadcast<float, long>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous
      namespace)::pythonic::operator_::functor::add, (anonymous namespace)::pythonic::types::broadcast<float, long>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::numpy::functor::exp, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous
      namespace)::pythonic::operator_::functor::neg, (anonymous namespace)::pythonic::types::numpy_gexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &, (anonymous namespace)::pythonic::types::contiguous_normalized_slice,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice> > > > >, 3, 0, false>' requested here
      return utils::broadcast_copy < numpy_gexpr &, E, value,
                    ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/types/numpy_gexpr.hpp:416:12: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::types::numpy_gexpr<const (anonymous namespace)::pythonic::types::ndarray<float,
      (anonymous namespace)::pythonic::types::pshape<long, long, long> > &, (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice>::_copy<(anonymous
      namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::div, (anonymous namespace)::pythonic::types::broadcast<float, long>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::add,
      (anonymous namespace)::pythonic::types::broadcast<float, long>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::numpy::functor::exp, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::neg,
      (anonymous namespace)::pythonic::types::numpy_gexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &, (anonymous namespace)::pythonic::types::contiguous_normalized_slice,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice> > > > > >' requested here
    return _copy(expr);
           ^
NERRNN/speedup.cpp:226:260: note: in instantiation of function template specialization '(anonymous namespace)::pythonic::types::numpy_gexpr<const (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice>::operator=<(anonymous namespace)::pythonic::types::numpy_expr<(anonymous
      namespace)::pythonic::operator_::functor::div, (anonymous namespace)::pythonic::types::broadcast<float, long>, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::add, (anonymous namespace)::pythonic::types::broadcast<float, long>,
      (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::numpy::functor::exp, (anonymous namespace)::pythonic::types::numpy_expr<(anonymous namespace)::pythonic::operator_::functor::neg, (anonymous namespace)::pythonic::types::numpy_gexpr<const
      (anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long, long> > &, (anonymous namespace)::pythonic::types::contiguous_normalized_slice, (anonymous namespace)::pythonic::types::contiguous_normalized_slice,
      (anonymous namespace)::pythonic::types::contiguous_normalized_slice> > > > > >' requested here
    conv_results(pythonic::types::contiguous_slice(pythonic::__builtin__::None,pythonic::__builtin__::None),pythonic::types::contiguous_slice(pythonic::__builtin__::None,pythonic::__builtin__::None),pythonic::types::contiguous_slice(state_size,state_size_4)) = (pythonic::operator_::div(1L, (pythonic::operator...
                                                                                                                                                                                                                                                                   ^
NERRNN/speedup.cpp:266:69: note: in instantiation of function template specialization '__pythran_speedup::conv::operator()<(anonymous namespace)::pythonic::types::ndarray<float, (anonymous namespace)::pythonic::types::pshape<long, long> > &, (anonymous namespace)::pythonic::types::ndarray<float,
      (anonymous namespace)::pythonic::types::pshape<long, long> > &, long &, long &, long &>' requested here
                                auto res = __pythran_speedup::conv()(conv_matrix, qcnn_filter, length, batch_size, state_size);
                                                                    ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/xsimd/types/xsimd_sse_float.hpp:79:11: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'decltype(xsimd::load_unaligned(this->data))' (aka 'xsimd::batch<double, 2>') to
      'const xsimd::batch<float, 4> &' for 1st argument
    class batch<float, 4> : public simd_batch<batch<float, 4>>
          ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/xsimd/types/xsimd_sse_float.hpp:79:11: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'decltype(xsimd::load_unaligned(this->data))' (aka 'xsimd::batch<double, 2>') to
      'xsimd::batch<float, 4> &&' for 1st argument
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/xsimd/types/xsimd_sse_float.hpp:248:29: note: candidate constructor not viable: no known conversion from 'decltype(xsimd::load_unaligned(this->data))' (aka 'xsimd::batch<double, 2>') to 'const __m128 &' for 1st argument
    inline batch<float, 4>::batch(const __m128& rhs)
                            ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/xsimd/types/xsimd_sse_double.hpp:95:9: note: candidate function
        operator __m128d() const;
        ^
/Users/lusheng.lls/.pyenv/versions/brm-index/lib/python3.7/site-packages/pythran/pythonic/include/types/nditerator.hpp:186:59: note: passing argument to parameter 'val' here
    void store(xsimd::simd_type<typename E::dtype> const &);
```

--- a/pythran/pythonic/types/numpy_gexpr.hpp
+++ b/pythran/pythonic/types/numpy_gexpr.hpp
@@ -379,18 +379,20 @@ namespace types
      * perform a fuzzy alias check dynamically!
      */
     assert(buffer);
+    constexpr bool vectorize =
+        is_vectorizable &&
+        std::is_same<dtype, typename dtype_of<E>::type>::value &&
+        is_vectorizable_array<E>::value;
     if (may_overlap(*this, expr)) {
       return utils::broadcast_copy<
           numpy_gexpr &, ndarray<typename E::dtype, typename E::shape_t>, value,
-          value - utils::dim_of<E>::value, is_vectorizable>(
+          value - utils::dim_of<E>::value, vectorize>(
           *this, ndarray<typename E::dtype, typename E::shape_t>(expr));
     } else {
       // 100% sure there's no overlap
-      return utils::broadcast_copy < numpy_gexpr &, E, value,
-             value - utils::dim_of<E>::value,
-             is_vectorizable &&
-                 std::is_same<dtype, typename dtype_of<E>::type>::value &&
-                 is_vectorizable_array<E>::value > (*this, expr);
+      return utils::broadcast_copy<numpy_gexpr &, E, value,
+                                   value - utils::dim_of<E>::value, vectorize>(
+          *this, expr);
     }
   }
 
@@ -400,13 +402,15 @@ namespace types
                           numpy_gexpr<Arg, S...> &>::type
   numpy_gexpr<Arg, S...>::_copy(E const &expr)
   {
+    constexpr bool vectorize =
+        is_vectorizable &&
+        std::is_same<dtype, typename dtype_of<E>::type>::value &&
+        is_vectorizable_array<E>::value;
     static_assert(value >= utils::dim_of<E>::value, "dimensions match");
     assert(buffer);
-    return utils::broadcast_copy < numpy_gexpr &, E, value,
-           value - utils::dim_of<E>::value,
-           is_vectorizable &&
-               std::is_same<dtype, typename dtype_of<E>::type>::value &&
-               is_vectorizable_array<E>::value > (*this, expr);
+    return utils::broadcast_copy<numpy_gexpr &, E, value,
+                                 value - utils::dim_of<E>::value, vectorize>(
+        *this, expr);
   }
 
   template <class Arg, class... S>

--- a/pythran/tests/cases/convnet.py
+++ b/pythran/tests/cases/convnet.py
@@ -1,0 +1,34 @@
+#pythran export convnet(float32[:,:], float32[:,:], int, int, int)
+#runas import numpy as np; x = np.arange(16., dtype=np.float32).reshape(4,4); y= np.arange(16., dtype=np.float32).reshape(4,4); convnet(x, y, 4,1,1)
+import numpy as np
+def sigmoid(z):
+    return 1 / (1 + np.exp(-z))
+
+
+def convnet(conv_matrix, qcnn_filter, length, batch_size, state_size):
+    state_size_2 = state_size * 2
+    state_size_3 = state_size * 3
+    state_size_4 = state_size * 4
+
+    conv_results = np.dot(conv_matrix, qcnn_filter).reshape(length, batch_size, state_size_4)
+    conv_results[:, :, :state_size] = np.tanh(conv_results[:, :, :state_size])
+    conv_results[:, :, state_size:state_size_4] = sigmoid(conv_results[:, :, state_size:state_size_4])
+
+    state_results = np.zeros((batch_size, length, state_size_2), dtype=np.float32)
+    state = np.zeros((batch_size, state_size), dtype=np.float32)
+    for i in range(length):
+        z = conv_results[i, :, :state_size]
+        f = conv_results[i, :, state_size:state_size_2]
+        o = conv_results[i, :, state_size_2:state_size_3]
+        state = f * state + (1 - f) * z
+        state_results[:, i, :state_size] = state * o
+
+    state = np.zeros((batch_size, state_size), dtype=np.float32)
+    for i in range(length - 1, -1, -1):
+        z = conv_results[i, :, :state_size]
+        f = conv_results[i, :, state_size_3:state_size_4]
+        o = conv_results[i, :, state_size_2:state_size_3]
+        state = f * state + (1 - f) * z
+        state_results[:, i, state_size:] = state * o
+    state_results = state_results.reshape((batch_size * length, state_size_2))
+    return state_results

